beamer/health/notify.py

```diff
@@ -0,0 +1,76 @@
+import json
+import time
+from typing import Callable, Optional, TypedDict
+
+import requests
+from typing_extensions import NotRequired
+
+
+_TIMEOUT_BETWEEN_SENDS = 61
+
+
+NotificationConfig = dict[str, dict[str, str]]
+
+
+class Message(TypedDict):
+    text: str
+    message_link: NotRequired[str]
+
+
+class Notify:
+    notifications_sent = 0
+
+    def __init__(self, system: str, config: NotificationConfig) -> None:
+        self.system = system
+        self.config = config
+
+    def send(self, message: Message, callback: Optional[Callable] = None) -> None:
+        if self.notifications_sent > 0 and self.notifications_sent % 5 == 0:
+            time.sleep(_TIMEOUT_BETWEEN_SENDS)
+
+        text = message["text"]
+
+        if message.get("message_link"):
+            text += "\n" + f"View on [explorer]({message['message_link']})"
+
+        if self.system == "rocketchat":
+            status = self.send_to_rocketchat(text)
+        elif self.system == "telegram":
+            status = self.send_to_telegram(text)
+        else:
+            raise ValueError("Unknown notification system")
```

Comment:
Nitpick: this is totally fine, but if you haven't know about the new `match` feature:

```python
match self.system:
        case "rocketchat":
            status = self.send_to_rocketchat(text)
        case "telegram":
            status = self.send_to_telegram(text)
        case _:
            raise ValueError("Unknown notification system")
```